utilities.cake

```diff
@@ -1339,6 +1346,271 @@ Task("verify-namespace-file")
+
+                if (line.Contains("Gradle"))
+                {
+                    tool = line.Replace("Gradle ", "");
+                    version = tool;
+                    sb.AppendLine($"Gradle, {version}");
+                }
+                else if (line.Contains("Kotlin:"))
+                {
+                    tool = line.Replace("Kotlin:", "").Trim();
+                    version = tool;
+                    sb.AppendLine($"Kotlin, {version}");
+                }
+                else if (line.Contains("Groovy:"))
+                {
+                    tool = line.Replace("Groovy:", "").Trim();
+                    version = tool;
+                    sb.AppendLine($"Groovy, {version}");
+                }
+                else if (line.Contains("Ant:"))
+                {
+                    tool = line.Replace("Ant:", "").Trim();
+                    version = tool;
+                    sb.AppendLine($"Ant, {version}");
+                }
+                else if (line.Contains("JVM:"))
+                {
+                    tool = line.Replace("JVM:", "").Trim();
+                    version = tool;
+                    sb.AppendLine($"JVM, {version}");
+                }
+            }
+            /*
+            java --version
+            */
+			process = "java";
+			process_args = "--version";
+            process_settings = new ProcessSettings ()
+			{
+                Arguments = process_args,
+                RedirectStandardOutput = true,
+            };
+			exitCodeWithoutArguments = StartProcess(process, process_settings, out redirectedStandardOutput);
+            foreach (string line in redirectedStandardOutput)
+            {
+                string tool = null;
+                string version = null;
+
+                if (line.Contains("openjdk"))
+                {
+                    tool = line.Replace("openjdk ", "");
+                    version = tool;
+                    sb.AppendLine($"openjdk, {version}");
+                }
+            }
+
+            /*
+            javac --version
+            */
+			process = "javac";
+			process_args = "--version";
+            process_settings = new ProcessSettings ()
+			{
+                Arguments = process_args,
+                RedirectStandardOutput = true,
+            };
+			exitCodeWithoutArguments = StartProcess(process, process_settings, out redirectedStandardOutput);
+            foreach (string line in redirectedStandardOutput)
+            {
+                string tool = null;
+                string version = null;
+
+                if (line.Contains("javac"))
+                {
+                    tool = line.Replace("javac ", "");
+                    version = tool;
+                    sb.AppendLine($"javac, {version}");
+                }
+            }
+
+			System.IO.File.WriteAllText("./output/buildtoolsinventory.csv", sb.ToString());
+			System.IO.File.WriteAllText("./docs/buildtoolsinventory.csv", sb.ToString());
```

Comment:
Could we upload this as a pipeline artifact so that the latest versions produced in CI can be downloaded? I didn't see it in the set of Published artifacts in the latest build.

Reply:
sure. let me see